### PR TITLE
[Global Clock] Clock, Clock Indicator, Timer and Notebook entries

### LINKED ISSUE #5630: Notebook entries should use Time API, not Date.now()
Currently notebook entries are timestamped with wall clock time. In simulation environments the "now" time may not necessarily correspond to the real-world time, in which case the notebook entries should be timestamped with the simulated time.

### PR DESCRIPTION
**Pull request by jvigliotta**
Closes #5630

### Describe your changes:

Added a `now` method to the TimeAPI as well as emitting a 'tick' event whenever the current clock ticks (regardless of time conductor mode). Modified the following components to use this:

- Notebook (entry timestamps)
- Clock
- Clock Indicator
- Timer

Also, modified our `raf` utility to accept arguments and deleted the now unused `ticker`.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Tests included and/or updated with changes?
* [ ] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [x] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [x] Code style and in-line documentation are appropriate?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: src/api/time/TimeContext.js [hunk 1]
```diff
@@ -371,6 +373,17 @@ class TimeContext extends EventEmitter {
...
 
+  /**
+   * Get the timestamp of the current clock
+   * @returns {number} current timestamp of current clock regardless of mode
+   * @memberof module:openmct.TimeAPI#
+   * @method now
+   */
+
+  now() {
+    return this.timestamp;
```
Comment (akhenry) [inline src/api/time/TimeContext.js:384]: I don't think we should be separately tracking the current timestamp in the time context. I think the sole source of time should be the clock, so let's remove the `timestamp` member variable and have this function call the active clock itself.

File: src/plugins/clock/components/Clock.vue [hunk 2]
```diff
@@ -85,12 +85,11 @@ export default {
     }
   },
   mounted() {
-    this.unlisten = ticker.listen(this.tick);
+    this.tick = raf(this.tick);
```
Comment (akhenry) [inline src/plugins/clock/components/Clock.vue:88]: yaaaas.

File: src/plugins/clock/components/ClockIndicator.vue [hunk 3]
```diff
@@ -46,16 +46,17 @@ export default {
     };
   },
   mounted() {
-    this.unlisten = ticker.listen(this.tick);
+    this.tick = raf(this.tick);
```
Comment (akhenry) [inline src/plugins/clock/components/ClockIndicator.vue:49]: This is a good change, thanks.

File: src/plugins/timer/actions/PauseTimerAction.js [hunk 4]
```diff
@@ -39,7 +39,7 @@ export default class PauseTimerAction {
 
     const newConfiguration = { ...domainObject.configuration };
     newConfiguration.timerState = 'paused';
-    newConfiguration.pausedTime = new Date();
+    newConfiguration.pausedTime = new Date(this.openmct.time.now());
```
Comment (akhenry) [inline src/plugins/timer/actions/PauseTimerAction.js:42]: The uses of `Date` here sets off my spidey senses.  The Time API supports time systems that are not directly parseable by JavaScript Dates (eg. SCET, LMST, etc.). That said, writing clocks and timers that support those time systems is a whole other deal...

File: src/api/time/TimeContext.js (hunk 1 above)
Comment (jvigliotta) [reply src/api/time/TimeContext.js:384]: Good point!

**Review by akhenry — APPROVED** [review]

File: src/utils/raf.js [hunk 5]
```diff
@@ -1,12 +1,12 @@
...
   let rendering = false;
 
-  return () => {
+  return (...args) => {
     if (!rendering) {
       rendering = true;
 
       requestAnimationFrame(() => {
-        callback();
+        callback(...args);
```
Comment (shefalijoshi) [inline src/utils/raf.js:9]: 👍

File: src/plugins/notebook/utils/notebook-entries.js [hunk 6]
```diff
@@ -159,7 +159,7 @@ export async function addNotebookEntry(
     return;
   }
 
-  const date = Date.now();
+  const date = openmct.time.now();
```
Comment (shefalijoshi) [inline src/plugins/notebook/utils/notebook-entries.js:162]: Can you confirm that notebook embeds are also using the clock time and not Date.now()?
Like when saving a snapshot?

**Review by shefalijoshi — CHANGES_REQUESTED** [review]
I see test failures too FYI.

**Review by shefalijoshi — APPROVED** [review]
NOICE

**Pull request merged**